Mosaic

weber8thomas opened the pull request:
Describe your pull request here

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

For members of the Bioconda project, the following command is also available:

<table>
  <tr>
    <td><code>@BiocondaBot please merge</code></td>
    <td>Upload built packages/containers and merge a PR. <br />Someone must approve a PR first! <br />This reduces CI build time by reusing built artifacts.</td>
  </tr>
</table>

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

weber8thomas commented:
@BiocondaBot please add label

bgruening commented on recipes/mosaic/meta.yaml:
```diff
@@ -0,0 +1,63 @@
+    - zlib
+    - bzip2
+
+# requirements:
```
Can you remove this if its not needed?

weber8thomas replied:
I just removed all the unnecessary commented parts in the last commit

weber8thomas commented:
@BiocondaBot please add label

weber8thomas commented:
@BiocondaBot please add label

jmarshall commented:
Please explain why the source URL and about:home refer to different GitHub repositories.

Also please explain why the bioconda package should be called `mosaic` rather than `mosaicatcher`.

jmarshall commented:
I didn't ask you to change it; I asked you to **explain** it. The canonical repository for this tool would appear to be [friendsofstrandseq/mosaic](https://github.com/friendsofstrandseq/mosaicatcher) (recently renamed from friendsofstrandseq/mosaicatcher, and still identifying itself as the longer name).

Normally a bioconda package would be built from the canonical repository for a tool. Why should it be built from your personal fork? Have you tried submitting your build changes to the upstream canonical repository? What changes did you need, and could they be done in the recipe's _build.sh_ instead?

There is already a bioconda package named `mosaik`. Ideally this new package would avoid confusion by differing by more than one character from the existing package. Hence the request for you to give us some insight into the multiple names used to refer to this tool.

I too expect that this does not depend directly on `zlib` and especially `bzip2`, which are implied by your `htslib` requirement.

weber8thomas commented:
> I didn't ask you to change it; I asked you to **explain** it. The canonical repository for this tool would appear to be [friendsofstrandseq/mosaic](https://github.com/friendsofstrandseq/mosaicatcher) (recently renamed from friendsofstrandseq/mosaicatcher, and still identifying itself as the longer name).

> Normally a bioconda package would be built from the canonical repository for a tool. Why should this be built from your personal fork? Have you tried submitting your build changes to the upstream canonical repository? What changes did you need, and could they be done in the recipe's _build.sh_ instead?

> There is already a bioconda package named `mosaik`. Ideally this new package would avoid confusion by differing by more than one character from the existing package. Hence the request for you to give us some insight into the multiple names used to refer to this tool.
> 
> I too expect that this does not depend directly on `zlib` and especially `bzip2`, which are implied by your `htslib` requirement.

Thank you for your reviews and sorry for the inconvenience. 
As I also had rights on the canonical repository, I made a PR and create a new tag dedicated to the bioconda version. 
The git repo name used to be `mosaicatcher` while the C++ program was named mosaic. For the sake of clarity and as you mentioned that a bioconda package named `mosaik` already existed, I kept the name `mosaicatcher` and changed it in the CmakeList file as well as in the README documentation. 
Finally I just removed `zlib` and `bzip2` from the requirements.

weber8thomas commented:
@bioconda/core : last commit checks passed & files are compliant to suggestions

bgruening reviewed (approved):
LGTM

weber8thomas commented:
@BiocondaBot please merge
